### PR TITLE
Fix crash on More Menu screen when site is not selected

### LINKED ISSUE #8718: IllegalStateException: SelectedSite.get() was accessed before being initialized - siteId -1.
Sentry Issue: [WOOCOMMERCE-ANDROID-6DQ](https://a8c.sentry.io/issues/4046104356/?referrer=github_integration)

```
IllegalStateException: SelectedSite.get() was accessed before being initialized - siteId -1.
Consider calling selectedSite.exists() to ensure site exists prior to calling selectedSite.get().
    at com.woocommerce.android.tools.SelectedSite.get(SelectedSite.kt:51)
    at com.woocommerce.android.ui.moremenu.domain.MoreMenuRepository.isUpgradesEnabled(MoreMenuRepository.kt:34)
    at com.woocommerce.android.ui.moremenu.MoreMenuViewModel.generateMenuButtons(MoreMenuViewModel.kt:101)
    at com.woocommerce.android.ui.moremenu.MoreMenuViewModel.access$getSelectedSiteAbsoluteUrl(MoreMenuViewModel.kt:34)
    at com.woocommerce.android.ui.moremenu.MoreMenuViewModel.access$generateMenuButtons(MoreMenuViewModel.kt:34)
...
(11 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by wzieba**
Closes: #8718

### Description

This PR fixes a crash which occured when user opened the More Menu screen with no selected site. I don't know how to reproduce this scenario but, looking at statistics of the bug in Sentry, this might be a corner case.

This PR should fix the problem.

### Testing instructions

As we don't have reproduction steps, there's no need for tests.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/domain/MoreMenuRepositoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import com.woocommerce.android.tools.SelectedSite
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.stub
+import org.wordpress.android.fluxc.model.SiteModel
+
+class MoreMenuRepositoryTest {
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/domain/MoreMenuRepositoryTest.kt:12]: Nice!

**Review by atorresveiga — APPROVED** [review]
Thanks for adding unit tests to explore all scenarios.
LGTM! :shipit:
